scripts/basic-workflow-tests.sh

```diff
@@ -253,48 +257,52 @@ test_taxonomy() {
 
 test_generate() {
     task Generate synthetic data
-    GENERATE_ARGS+=("--endpoint-url" "http://localhost:8000/v1")
+    local generate_args
+    
+    generate_args=("--endpoint-url" "http://localhost:8000/v1")
     if [ "$MIXTRAL" -eq 1 ]; then
-        GENERATE_ARGS+=("--model" "${CACHE_HOME}/instructlab/models/${MIXTRAL_GGUF_MODEL}")
+        generate_args+=("--model" "${CACHE_HOME}/instructlab/models/${MIXTRAL_GGUF_MODEL}")
     else
-        GENERATE_ARGS+=("--model" "${CACHE_HOME}/instructlab/models/${MERLINITE_GGUF_MODEL}")
+        generate_args+=("--model" "${CACHE_HOME}/instructlab/models/${MERLINITE_GGUF_MODEL}")
     fi
 
     # default arg for '--pipeline' is "simple"
     if [ "$SDG_PIPELINE" = "full" ]; then
-        GENERATE_ARGS+=("--pipeline" "full")
+        generate_args+=("--pipeline" "full")
     fi
 
     # Disable batching with llama-cpp. See https://github.com/instructlab/instructlab/issues/1892
     if [ "$BACKEND" = "llama-cpp" ]; then
-        GENERATE_ARGS+=("--batch-size" "0")
+        generate_args+=("--batch-size" "0")
     fi
 
     # NUM_INSTRUCTIONS is '1' if MINIMAL is set, '5' otherwise
-    ilab data generate --num-instructions ${NUM_INSTRUCTIONS} "${GENERATE_ARGS[@]}"
+    ilab data generate --num-instructions ${NUM_INSTRUCTIONS} "${generate_args[@]}"
 }
 
 test_train() {
     task Train the model
+    local train_args
 
     if [ "$TRAIN_LIBRARY" -eq 1 ]; then
-        DATA=$(find "${DATA_HOME}"/instructlab/datasets -name 'messages_*' | head -n 1)
+        local data
+        data=$(find "${DATA_HOME}"/instructlab/datasets -name 'messages_*' | head -n 1)
         # TODO Only cuda for now
-        # the train profile specified in test_init overrides the majority of TRAIN_ARGS, including things like num_epochs. While it looks like much of those settings are being lost, they just have different values here.
-        TRAIN_ARGS=("--device=cuda" "--model-path=${GRANITE_SAFETENSOR_REPO}" "--data-path=${DATA}" "--lora-quantize-dtype=nf4" "--4-bit-quant" "--effective-batch-size=4" "--is-padding-free=False")
+        # the train profile specified in test_init overrides the majority of train_args, including things like num_epochs. While it looks like much of those settings are being lost, they just have different values here.
+        train_args=("--device=cuda" "--model-path=${GRANITE_SAFETENSOR_REPO}" "--data-path=${data}" "--lora-quantize-dtype=nf4" "--4-bit-quant" "--effective-batch-size=4" "--is-padding-free=False")
         if [ "${BACKEND}" != "vllm" ]; then
-            TRAIN_ARGS+=("--gguf-model-path" "${CACHE_HOME}/instructlab/models/${GRANITE_GGUF_MODEL}")
+            train_args+=("--gguf-model-path" "${CACHE_HOME}/instructlab/models/${GRANITE_GGUF_MODEL}")
         fi
 
         ilab model train "${TRAIN_ARGS[@]}"
     else
         # TODO Only cuda for now
-        TRAIN_ARGS+=("--legacy" "--device=cuda")
+        train_args=("--legacy" "--device=cuda")
```

Comment:
This was appended to the global TRAIN_ARGS which is still there.  It's also appended to in set_defaults.  Maybe this one shouldn't switch to a local?

Reply:
Yeah you're right, I'm going to leave this one be